Mathlib/GroupTheory/Coxeter/Basic.lean

```diff
@@ -56,162 +49,119 @@ presentation corresponding to a Coxeter matrix which is registered in a Coxeter
 
-end CoxeterGroup
+/-- The isomorphism between the Coxeter group associated to the reindexed matrix `M.reindex e` and
+the Coxeter group associated to `M`. -/
+def reindexGroupEquiv : (M.reindex e).group ≃* M.group :=
+  (QuotientGroup.congr (Subgroup.normalClosure M.relationsSet)
+    (Subgroup.normalClosure (M.reindex e).relationsSet)
+    (FreeGroup.freeGroupCongr e) (by
+      rw [reindex_relationsSet,
+        Subgroup.map_normalClosure _ _ (FreeGroup.freeGroupCongr e).surjective,
+        ← MulEquiv.coe_toMonoidHom]
+      rfl)).symm
 
-/-- The group presentation corresponding to a Coxeter matrix. -/
-def Matrix.CoxeterGroup := PresentedGroup <| CoxeterGroup.Relations.toSet M
+theorem reindexGroupEquiv_apply_simple (i : B') :
+    (M.reindexGroupEquiv e) ((M.reindex e).simple i) = M.simple (e.symm i) := rfl
 
-instance : Group (Matrix.CoxeterGroup M) :=
-  QuotientGroup.Quotient.group _
+theorem reindexGroupEquiv_symm_apply_simple (i : B) :
+    (M.reindexGroupEquiv e).symm (M.simple i) = (M.reindex e).simple (e i) := rfl
 
-namespace CoxeterGroup
+end CoxeterMatrix
 
-/-- The canonical map from `B` to the Coxeter group with generators `b : B`. The term `b`
-is mapped to the equivalence class of the image of `b` in `CoxeterGroup M`. -/
-def of (b : B) : Matrix.CoxeterGroup M := PresentedGroup.of b
+/-! ### Coxeter systems -/
 
-@[simp]
-theorem of_apply (b : B) : of M b = PresentedGroup.of (rels := Relations.toSet M) b :=
-  rfl
+section
 
-end CoxeterGroup
+variable {B : Type*} (M : CoxeterMatrix B)
 
-/-- A Coxeter system `CoxeterSystem W` is a structure recording the isomorphism between
-a group `W` and the group presentation corresponding to a Coxeter matrix. Equivalently, this
-can be seen as a list of generators of `W` parameterized by the underlying type of `M`, which
-satisfy the relations of the Coxeter matrix `M`. -/
-structure CoxeterSystem (W : Type*) [Group W]  where
-  /-- `CoxeterSystem.ofMulEquiv` constructs a Coxeter system given an equivalence with the group
-  presentation corresponding to a Coxeter matrix `M`. -/
-  ofMulEquiv ::
-    /-- `mulEquiv` is the isomorphism between the group `W` and the group presentation
-    corresponding to a Coxeter matrix `M`. -/
-    mulEquiv : W ≃* Matrix.CoxeterGroup M
+/-- A Coxeter system `CoxeterSystem M W` is a structure recording the isomorphism between
+a group `W` and the Coxeter group associated to a Coxeter matrix `M`. -/
+@[ext]
+structure CoxeterSystem (W : Type*) [Group W] where
+  /-- The isomorphism between `W` and the Coxeter group associated to `M`. -/
+  mulEquiv : W ≃* M.group
 
 /-- A group is a Coxeter group if it admits a Coxeter system for some Coxeter matrix `M`. -/
-class IsCoxeterGroup (W : Type u) [Group W] : Prop where
-  nonempty_system : ∃ B : Type u, ∃ M : Matrix B B ℕ, M.IsCoxeter ∧ Nonempty (CoxeterSystem M W)
+class IsCoxeterGroup (W : Type*) [Group W] : Prop where
+  nonempty_system : ∃ B : Type*, ∃ M : CoxeterMatrix B, Nonempty (CoxeterSystem M W)
```

Comment:
This looks wrong to me; I'm pretty sure this is defining a typeclass `IsCoxeterGroup.{u,v} W`, which has more universe variables than expected. If this is deliberate, then I think writing the variables explicitly would be better.